providers.d/apt-mirror/provider_wrapper.sh

```diff
@@ -1,13 +1,37 @@
 bashelliteProviderWrapper() {
 
-  if [[ ! ${_r_dryrun} ]]; then
-    apt-mirror ${_r_metadata_tld}/repos.conf.d/${_n_repo_name}/provider.conf;
+  local basepath_parameter="$(grep -oP "(?<=(^set base_path[[:space:]]*))[[:alnum:]/]*" ${_r_metadata_tld}/repos.conf.d/${_n_repo_name}/provider.conf)"
+  if [[  "${basepath_parameter}" != "${_r_mirror_tld}/${_n_repo_name}" ]]; then
+    utilMsg WARN "$(utilTime)" "The \"base_path\" parameter (${basepath_parameter}) in provider.conf does not match mirror location (${_r_mirror_tld}/${_n_repo_name})..."
   fi
 
-  if [[ "${?}" != "0" ]]; then
-    utilMsg WARN "$(utilTime)" "apt-mirror either failed or completed with errors for repo (${_n_repo_name})."
+  local config_file="${_r_metadata_tld}/repos.conf.d/${_n_repo_name}/provider.conf"
+  local tmp_config_file="${HOME}/.bashellite/tmp_${_n_repo_name}_provider.conf"
+
+  cat /dev/null > ${tmp_config_file}
+
+  IFS=$'\n'
+  for line in $(cat ${config_file}); do
+    if [[ ${line} =~ ^set[[:blank:]]base_path[[:blank:]]+[[:alnum:]/]+ ]]; then
+      echo "set base_path         ${_r_mirror_tld}/${_n_repo_name}" >> ${tmp_config_file}
+    else
+      echo "${line}" >> ${tmp_config_file}
```

Comment:
Is the config file valid if this is dropped at the bottom of the file, or does it need to be done inline? Should this be a sed instead?

Reply:
This is done inline and not appended at the end of the file.  The current for loop take each line from the original config file and echos it to the tmp file replacing the line with "set base_path" with one that has the current mirror_tld/repo_name used for the execution.

Reply:
I didn't notice this initially because I was only looking at the changed lines. After looking at the file stand-alone, your methodology is apparent. This is a nice way to do it that does not require an external stream editor. A potential improvement might be adding a ${counter} variable or some other mechanism to this that lets you detect how many times this piece of code has been called. That way, you can handle subsequent calls differently (by removing them). For example, the first time the script encounters base_path, it updates the value and every other time, it removes the line completely so that only one line appears in the new file. You could even add a ${current_section} variable for ini-style files that ensures that the parameter is placed under the appropriate [ini_section_header]. The logic/pseudo-code might look something like:

```
line_counter=0;
for line in $lines_in_file; do
 
  if $line == regex_that_detects_ini_headers; then
     current_ini_section=$line;
  fi

  if $line +~ regex_that_detects_parameter; then
    if $current_ini_section="desired_section" && $line_counter == 0; then
       echo "parameter=new_parameter_value"
       line_counter+=1;
    else
       # if under wrong section, or duplicate, remove duplicate/misplaced line from file completely
    fi
done
```

Reply:
Updated.